raincloudy/helpers.py

```diff
@@ -12,49 +12,70 @@ def generate_soup_html(data):
         raise
 
 
-def serial_finder(data):
+def faucet_serial_finder(data):
     """
-    Find controller serial and faucet_serial from the setup page.
+    Find faucet_serial from the setup page.
 
     <select id="id_select_controller2" name="select_controller" >
         <option value='0' selected='selected'>1 - Controller001</option>
+        <option value='1>2 - Controller002</option>
     </select>
 
     :param data: text to be parsed
     :type data: BeautilSoup object
-    :return: a dict with controller_serial and faucet_serial
-    :rtype: dict
+    :return: a dict with array of controller_serials and array of faucet_serials
     :raises IndexError: if controller_serial was not found on the data
     """
     if not isinstance(data, BeautifulSoup):
         raise TypeError("Function requires BeautifulSoup HTML element.")
 
     try:
 
-        # The setup page contains a select box for each controller and each
-        # faucet
-        controllersElement = data.find_all('select',
-                                           {'id': 'id_select_controller2'})
+        faucets_element = data.find(id="id_select_faucet2").find_all('option')
 
-        faucetsElement = data.find_all('select',
-                                       {'id': 'id_select_faucet2'})
+        faucet_serials = []
+        for faucet_element in faucets_element:
+            faucet_serials.append(faucet_element.text.split('-')[1].strip())
 
-        controllerSerial = controllersElement[0].text.split('-')[1].strip()
-        faucetSerial = faucetsElement[0].text.split('-')[1].strip()
-
-        # currently only one faucet is supported on the code
-        # we have plans to support it in the future
-        parsed_dict = {}
-        parsed_dict['controller_serial'] = controllerSerial
-        parsed_dict['faucet_serial'] = [faucetSerial]
-        return parsed_dict
+        return faucet_serials
 
     except (AttributeError, IndexError, ValueError):
         raise RainCloudyException(
             'Could not find any valid controller or faucet')
 
 
-def find_controller_or_faucet_name(data, p_type):
+def controller_serial_finder(data):
```

Comment:
Will likely recombine this and the faucet function to behave like the name find function does